### ISSUE TITLE
🚀 [Feature]: Ctx made into an interface

### ISSUE THREAD

**Issue by pjebs** [gofiber/fiber#2226]
### Feature Description

1. Currently Ctx is a *struct that **does not contain** any exported fields.
2. Can it be made into **an interface** instead just like the `Ctx` in standard library.
3. Since Ctx contains only methods, the change is backward compatible.

**Reasons:**

Developers should be permitted to override methods.

### Additional Context (optional)

In my case, I need to be able to override: `GetRouteURL()`

### Checklist:

- [X] I agree to follow Fiber's [Code of Conduct](https://github.com/gofiber/fiber/blob/master/.github/CODE_OF_CONDUCT.md).
- [X] I have checked for existing issues that describe my suggestion prior to opening this one.
- [X] I understand that improperly formatted feature requests may be closed without explanation.

**Comment by efectn** [comment]
It was already done in `v3-beta` branch. It's breaking change for v2.

**Comment by pjebs** [comment]
When is v3 expected to be released?

**Comment by pjebs** [comment]
Also how is it not backward compatible?

**Comment by efectn** [comment]
> Also how is it not backward compatible?

It means a change in handler signature and it's not really backward-compatible

**Comment by pjebs** [comment]
where is the change in handler signature? How is it backward incompatible?

**Comment by efectn** [comment]
Before: `func(*fiber.Ctx) error`
After: `func(fiber.Ctx) error`
